FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+import { Link } from 'react-router-dom';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { useState, useMemo } from 'react';
+import { useSWRConfig } from 'swr';
+import useSWRMutation from 'swr/mutation';
+import styles from './CompanyCard.module.css';
+import { useUser } from '../../hooks';
+import PropTypes from 'prop-types';
+import { Tooltip, Badge } from 'antd';
+
+export default function CompanyCard({ data, isAuthorized }) {
+  CompanyCard.propTypes = {
+    data: PropTypes.object,
+    isAuthorized: PropTypes.bool,
+  };
+  const [isSaved, setIsSaved] = useState(data && data.is_saved);
```

Comment:
it's better to pass a callback as a prop to update `data` prop to follow the single responsibility principle

Reply:
Can you explain what do you mean?